File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/2830.py
a = int(input())
b = int(input())

if((a <= 8 and b >= 9) or (b <= 8 and a >= 9)):
    print("final")
elif((a % 2 == 0 and b == a-1) or (b % 2 == 0 and a == b-1)):
    print("oitavas")
elif((a <= 12 and b >= 13) or (a >= 13 and b <= 12) or (a <= 4 and b >= 5) or (a >= 5 and b <= 4)):
    print("semifinal")
else:
    print("quartas")

File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/2830.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8')
var lines = input.split('\n')

    
    a = parseInt(lines.shift())
    b = parseInt(lines.shift())
    
    if((a <= 8 && b >= 9) || (b <= 8 && a >= 9)){
        console.log("final");
    } else if((a % 2 == 0 && b == a-1) || (b % 2 == 0 && a == b-1)){
        console.log("oitavas");
    } else if((a <= 12 && b >= 13) || (a >= 13 && b <= 12) || (a <= 4 && b >= 5) || (a >= 5 && b <= 4)){
        console.log("semifinal");
    } else{
        console.log("quartas");
    }

